integration/rust_signer/src/main.rs

```diff
@@ -0,0 +1,55 @@
+use bbs::prelude::*;
+use std::env;
+use std::convert::TryInto;
+
+fn parse_sig_bytes<T>(v: Vec<T>) -> [T; 112] {
+    v.try_into()
+        .unwrap_or_else(|v: Vec<T>| panic!("Expected a Vec of length {} but it was {}", 112, v.len()))
+}
+
+fn parse_pk_bytes<T>(v: Vec<T>) -> [T; 96] {
+    v.try_into()
+        .unwrap_or_else(|v: Vec<T>| panic!("Expected a Vec of length {} but it was {}", 96, v.len()))
+}
+
+fn main() {
+    let messages = vec![
+        SignatureMessage::hash(b"message 1"),
+        SignatureMessage::hash(b"message 2"),
+        SignatureMessage::hash(b"message 3"),
+        SignatureMessage::hash(b"message 4"),
+        SignatureMessage::hash(b"message 5"),
+    ];
+
+    let args: Vec<String> = env::args().collect();
+
+    if args.len() == 1 {
```

Comment:
so, this branch is signing, the other branch is verifying?

Reply:
correct